adsplanetnamepipe/app.py

```diff
@@ -0,0 +1,437 @@
+                return rows_updated
+
+            except SQLAlchemyError as e:
+                session.rollback()
+                self.logger.error(f"Error occurred while appending keyword `{keyword.lower()}` for feature name `{feature_name_entity}` and target `{target_entity}` for the `KnowledgeBase` records: {str(e)}")
+                return -1
+
+    def remove_from_knowledge_base_keywords(self, feature_name_entity, target_entity, keyword):
+        """
+        remove the keyword from the keywords for a specific feature name/target
+        note that this is only implemented for the `planetary` records
+
+        :param feature_name_entity:
+        :param target_entity:
+        :param keyword:
+        :return: number of rows updated
+        """
+        with self.session_scope() as session:
+            try:
+                knowledge_base_history = session.query(KnowledgeBaseHistory.id.label('id')) \
+                    .filter(and_(KnowledgeBaseHistory.feature_name_entity == feature_name_entity,
+                                 KnowledgeBaseHistory.target_entity == target_entity,
+                                 KnowledgeBaseHistory.named_entity_label == 'planetary')) \
+                    .subquery()
+
+                rows_updated = session.query(KnowledgeBase) \
+                    .filter(and_(KnowledgeBase.history_id == knowledge_base_history.c.id,
+                                 KnowledgeBase.keywords.op('@>')('{' + keyword.lower() + '}'))) \
+                    .update(
+                        {KnowledgeBase.keywords: func.array_remove(KnowledgeBase.keywords, keyword.lower())},
+                        synchronize_session='fetch'
+                    )
+                session.commit()
+
+                self.logger.info(f"{rows_updated} rows updated by removing the keyword `{keyword.lower()}` for feature name `{feature_name_entity}` and target `{target_entity}` from the `KnowledgeBase` records.")
+                return rows_updated
+
+            except SQLAlchemyError as e:
+                session.rollback()
+                self.logger.error(f"Error occurred while removing keyword `{keyword.lower()}` for feature name `{feature_name_entity}` and target `{target_entity}` from the `KnowledgeBase` records: {str(e)}")
+                return -1
+
+    def get_most_recent_knowledge_base_history_records(self, feature_name_entity, target_entity):
+        """
+
+        :param feature_name_entity:
+        :param target_entity:
+        :return:
+        """
+        with self.session_scope() as session:
+            rows = session.query(KnowledgeBaseHistory) \
+                .filter(and_(KnowledgeBaseHistory.feature_name_entity == feature_name_entity,
+                             KnowledgeBaseHistory.target_entity == target_entity,
+                             KnowledgeBaseHistory.named_entity_label == 'planetary')) \
```

Comment:
Consider using the lookup table here, since you're using one

Reply:
Not follow. Please elaborate.

Reply:
I was thinking along the lines of doing a join on the `NamedEntityLabel` table, on `KnowledgeBaseHistory.named_entity_label == NamedEntityLabel.label`

Reply:
The join with the NamedEntityLabel table is not necessary for this query. Since there's a foreign key constraint between KnowledgeBaseHistory.named_entity_label and NamedEntityLabel.label, we don't need to explicitly join these tables for this operation. The foreign key ensures data integrity at the database level. Also since we're only filtering on the label value, which is directly stored in KnowledgeBaseHistory, we can use the simpler query without the join.